proto/network.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+package proto;
+option go_package = ".;proto";
+
+import "session.proto";
+
+service Network {
+    rpc HTTPRequest (HTTPRequestMsg) returns (HTTPResponseMsg);
+}
+
+message HTTPRequestMsg {
+    Session session = 1;
+    string url = 2;
+    string method = 3;
+    bytes body = 4;
```

Comment:
As discussed this should include a way for consumers to supply (and read) headers, but I understand this will be implemented in a subsequent PR.

Reply:
https://crosschx.atlassian.net/browse/SIDE-969